source/adios2/engine/dataman/DataManReader.cpp

```diff
@@ -101,8 +100,21 @@ StepStatus DataManReader::BeginStep(StepMode stepMode,
                   << m_CurrentStep << std::endl;
     }
 
-    if (m_InitFailed or (m_CurrentStep == m_FinalStep && m_CurrentStep > 0))
+    if (m_InitFailed)
     {
+        if (m_Verbosity >= 5)
+        {
+            std::cout << "DataManReader::BeginStep() returned EndOfStream due to initialization failure" << std::endl;
```

Comment:
Bikeshedding: Shouldn't this go to `std::cerr` and replace the `std::endl` with a `\n` to avoid the buffer flush?

Reply:
Short answer, I don't know. Maybe you are right. Long answer, in principle yes, this is some kind of error. But the only reason I do this here is for debug purpose. If I have some of the information go to std::cout and other go to std::cerr, then it will mess up the sequence of the events over time, and it's much harder for anyone to make sense out of it. For the same reason, I prefer a buffer flush because I want the outputs to reflect the exact time sequence across all processes / threads for every event that happened.

Reply:
@JasonRuonanWang : That all sounds very sensible to me; sorry for the noise!